app/models/reports/lsa/fy2019/base.rb

```diff
@@ -70,39 +72,238 @@ def value_for_options options
         if options['project_id'].is_a?(Array)
           if options['project_id'].delete_if(&:blank?).any?
-            str = "; Projects: #{options['project_id'].map{|m| GrdaWarehouse::Hud::Project.find_by_id(m.to_i)&.name || m if m.present?}.compact.join(', ')}"
+            str = "; Projects: #{options['project_id'].map do |m|
+              GrdaWarehouse::Hud::Project.find_by_id(m.to_i)&.name || m if m.present? # rubocop:disable Metrics/BlockNesting
+            end.compact.join(', ')}"
           end
         else
-          str = "; Project: #{GrdaWarehouse::Hud::Project.find_by_id(options['project_id'].to_i)&.name || options['project_id'] }"
+          str = "; Project: #{GrdaWarehouse::Hud::Project.find_by_id(options['project_id'].to_i)&.name || options['project_id']}"
         end
       end
-      return str
+      str
     end
 
-    def project_group_string options
+    private def project_group_string options
       if (pg_ids = options['project_group_ids']&.compact) && pg_ids&.any?
         names = GrdaWarehouse::ProjectGroup.where(id: pg_ids).pluck(:name)
-        if names.any?
-          return "; Project Groups: #{names.join(', ')}"
-        end
+        return "; Project Groups: #{names.join(', ')}" if names.any?
       end
       ''
     end
 
-    def sub_population_string options
+    private def sub_population_string options
       if (sub_population = options['sub_population']) && sub_population.present?
         return "; Sub Population: #{sub_population.humanize.titleize}"
       end
       ''
     end
 
+    def missing_data(user)
+      @missing_data = {}
+      @range = ::Filters::DateRange.new(start: Date.current - 3.years, end: Date.current)
+      @missing_data[:missing_housing_type] = add_missing_housing_types(user)
+      @missing_data[:missing_geocode] = add_missing_geocodes(user)
+      @missing_data[:missing_geography_type] = add_geography_types(user)
+      @missing_data[:missing_zip] = add_missing_zips(user)
+      @missing_data[:missing_operating_start_date] = add_operating_start_dates(user)
+      @missing_data[:invalid_funders] = add_invalid_funders(user)
+      @missing_data[:missing_coc_codes] = add_missing_coc_codes(user)
+      @missing_data[:missing_inventory_coc_codes] = add_missing_inventory_coc_codes(user)
+      @missing_data[:missing_inventory_start_dates] = add_missing_inventory_start_dates(user)
+
+      @missing_data[:missing_projects] = @missing_data.values.flatten.uniq.sort_by(&:first)
+      @missing_data[:show_missing_data] = @missing_data[:missing_projects].any?
+      @missing_data
+    end
 
+    private def missing_data_columns
+      {
+        project_name: p_t[:ProjectName].to_sql,
+        org_name: o_t[:OrganizationName].to_sql,
+        project_type: p_t[:computed_project_type].to_sql,
+        funder: f_t[:Funder].to_sql,
+        id: p_t[:id].to_sql,
+        ds_id: p_t[:data_source_id].to_sql,
+      }
+    end
+
+    private def add_missing_housing_types(user)
```

Comment:
I'd take the `add` out of these names, these don't actually add anything?
Also, I don't know if it worth it, but except for the query, are these all identical? Maybe DRY them up?